x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -61,6 +62,121 @@ func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx
 	}, nil
 }
 
+func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCreatePosition, error) {
+	// get random pool
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get random user address with the pool denoms
+	sender, tokens, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	// ensure that we always have 2 tokens
+	// Note: tokens returns a random subset of poolDenoms, so  had to add this assertion
+	if len(tokens) < 2 {
+		return nil, fmt.Errorf("user doesnot have pool tokens")
+	}
+
+	//  Retrieve minTick and maxTick from precision factor
+	minTick, maxTick := clkeeper.GetMinAndMaxTicksFromExponentAtPriceOne(clPool.GetPrecisionFactorAtPriceOne())
+
+	// Randomize lowerTick and upperTick from max values to create position
+	lowerTick, upperTick, err := getRandomTickPositions(sim, minTick, maxTick, clPool.GetTickSpacing())
+	if err != nil {
+		return nil, err
+	}
+
+	return &cltypes.MsgCreatePosition{
+		PoolId:        clPool.GetId(),
+		Sender:        sender.Address.String(),
+		LowerTick:     lowerTick,
+		UpperTick:     upperTick,
+		TokenDesired0: tokens[0],
+		TokenDesired1: tokens[1],
+		// TODO: Randomize TokenMinAmount0 and TokenMinAmount1 in next iteration
+		TokenMinAmount0: sdk.NewInt(0),
+		TokenMinAmount1: sdk.NewInt(0),
+	}, nil
+}
+
+func RandMsgWithdrawPosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgWithdrawPosition, error) {
+	rand := sim.GetSeededRand("select random position")
+	// get random pool
+	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get random user address with the pool denoms
+	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	positions, err := k.GetUserPositions(ctx, sender.Address)
+	if err != nil {
+		return nil, fmt.Errorf("position does not exist")
+	}
+
+	if len(positions) == 0 {
+		return nil, fmt.Errorf("user does not have any position")
+	}
+
+	// pick a random position
+	randPosition := positions[rand.Intn(len(positions))]
+
+	// get percentage amount from 1 to 100 to withdraw liquidity
+	randPerc := sdk.MustNewDecFromStr(fmt.Sprintf("%.2f", sim.RandomDecAmount(sdk.OneDec())))
```

Comment:
nit: please double check but I think this should be sufficient for getting the desired value
```suggestion
	randPerc := sim.RandomDecAmount(sdk.OneDec())
```

Reply:
Confirmed this is working